environments/nodejs/server.js

```diff
@@ -7,56 +7,73 @@ const express = require('express');
 const app = express();
 const bodyParser = require('body-parser');
 const morgan = require('morgan');
+const argv = require('minimist')(process.argv.slice(1));// Command line opts
 
-// Command line opts
-const argv = require('minimist')(process.argv.slice(1));
-if (!argv.codepath) {
-    argv.codepath = "/userfunc/user";
-    console.log("Codepath defaulting to ", argv.codepath);
-}
 if (!argv.port) {
-    console.log("Port defaulting to 8888");
     argv.port = 8888;
 }
 
-
-// Node resolves module paths according to a file's location. We load
-// the file from argv.codepath, but tell users to put dependencies in
-// the server's package.json; this means the function's dependencies
-// are in /usr/src/app/node_modules.  We could be smarter and have the
-// function deps in the right place in argv.codepath; but for now we
-// just symlink the function's node_modules to the server's
-// node_modules.
-fs.symlinkSync('/usr/src/app/node_modules', `${path.dirname(argv.codepath)}/node_modules`);
-
 // User function.  Starts out undefined.
 let userFunction;
 
-//
-// Specialize this server to a given user function.  The user function
-// is read from argv.codepath; it's expected to be placed there by the
-// fission runtime.
-//
-function specialize(req, res) {
-    // Make sure we're a generic container.  (No reuse of containers.
-    // Once specialized, the container remains specialized.)
-    if (userFunction) {
-        res.status(400).send("Not a generic container");
-        return;
-    }
-
+function loadFunction(modulepath, funcname) {
     // Read and load the code. It's placed there securely by the fission runtime.
     try {
         var startTime = process.hrtime();
-        userFunction = require(argv.codepath);
+        // support v1 codepath and v2 entrypoint like 'foo', '', 'index.hello'
+        userFunction = funcname ? require(modulepath)[funcname] : require(modulepath);
         var elapsed = process.hrtime(startTime);
         console.log(`user code loaded in ${elapsed[0]}sec ${elapsed[1]/1000000}ms`);
+        return true;
     } catch(e) {
         console.error(`user code load error: ${e}`);
-        res.status(500).send(JSON.stringify(e));
-        return;
+        return false;
+    }
+}
+
+function withMakesureGeneric(func) {
+    return function(req, res) {
+        // Make sure we're a generic container.  (No reuse of containers.
+        // Once specialized, the container remains specialized.)
+        if (userFunction) {
+            res.status(400).send("Not a generic container");
+            return;
+        }
+
+        func(req, res);
     }
-    res.status(202).send();
+}
+
+function specializeV2(req, res) {
+    // for V2 entrypoint, 'filename.funcname' => ['filename', 'funcname']
+    const entrypoint = req.body.functionName ? req.body.functionName.split('.') : [];
+    // for V2, filepath is dynamic path
```

Comment:
check entrypoint.length and make sure it's what you expect, otherwise return an informative error

Reply:
I'm not sure this is the better way as the entrypoint can be empty(default is index.js) or a file name or `filename.functionname`

Reply:
I see, ok.